### ISSUE TITLE
Accordion | wrong aria roles and add headerAriaLevel property (accessibility)

### ISSUE THREAD

**Issue by GiacomoDM** [primefaces/primeng#13431]
### Describe the bug

As per W3C specification, the accordion should be a sequence of headers with buttons. 
In the [Accessibility section of PrimeNG's Accordion documentation](https://primeng.org/accordion#accessibility) is stated that 
> Accordion header elements have a button role

but actually the accordion still has `role="tablist"` and each header has `role="tab"`. In PrimeVue the accordion is not a `tablist` and the `a` tag has role `button`, correctly. 

Furthermore, it is recommended in the [WAI ARIA pattern for Accordions](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/#wai-ariaroles,states,andproperties:) that each header button should be wrapped in a container with `role="heading"`. Such role also requires `aria-level` to be defined or [it will assume the default value of `2` ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role#associated_wai-aria_roles_states_and_properties).

### Environment

Windows

### Angular version

16.X

### PrimeNG version

16.1.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.20

### Steps to reproduce the behavior

Open https://primeng.org/accordion and check the aria roles of the accordion's HTML structure.

### Expected behavior

Same as PrimeVue's one, plus having the `div` that wraps the `a` tag to have `role="heading"` and a customizable `aria-level`.
